File: bishi/meituan02.cpp
#include<iostream>
#include<vector>
using namespace std;

int main(){
    int n=0;
    cin>>n;
    vector<int> bef(n);
    vector<int> aft(n);
    int a=0;
    for(int i=0;i<n;i++){
        cin>>a;
        bef[i]=a;
    }
    for(int i=0;i<n;i++){
        cin>>a;
        aft[i]=a;
    }
    int count=0;
    vector<int> index(n);
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if(bef[i]==aft[j]){
                index[i]=j;
                break;
            }
        }
    }
    for(int i=n-1;i>0;i--){
        for(int j=i-1;j>=0;j--){
            if(index[i]<index[j]){
                count++;
                break;
            }
        }
    }
    cout<<count<<endl;
   return 0;
}

File: bishi/meituan02.js
// var n=parseInt(readline());
// var before=readline().split(' ').map(Number);
// var after=readline().split(' ').map(Number);
var n=5;
var before=[5,3,1,4,2];
var after=[2,4,5,1,3];

function a(n,bef,aft){
    var index=[];
    var index1=0;
    var count=0;
    for(let i=0;i<n;i++){
         index1=aft.indexOf(bef[i]);
         index.push(index1);
    }
    for(let i=n-1;i>0;i--){
        for(let j=i-1;j>=0;j--){
            if(index[i]<index[j]){
                count++;
                break;
            }
        }
    }
    return count;
}
console.log(a(n,before,after));